File: ios/DownloadManagerKit/Sources/DownloadManagerKit/DownloadStatus.swift
//
//  DownloadStatus.swift
//  DownloadManagerKit
//

/// Represents the various states of a download item.
public enum DownloadStatus: String, Codable, CaseIterable, Sendable {
   /// Status could not be determined.
   case unknown
   /// Download has not yet been created/persisted.
   case pending
   /// Download has been created and is ready to start.
   case idle
   /// Download is in progress.
   case inProgress
   /// Download was in progress but has been paused.
   case paused
   /// Download was canceled by the user.
   case canceled
   /// Download completed.
   case completed
}

File: android/lib/src/main/java/org/silvermine/downloadmanager/DownloadStatus.kt
package org.silvermine.downloadmanager

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Represents the various states of a download item.
 */
@Serializable
enum class DownloadStatus {
   /** Download has been created and is ready to start. */
   @SerialName("idle")
   Idle,

   /** Download is in progress. */
   @SerialName("inProgress")
   InProgress,

   /** Download was in progress but has been paused. */
   @SerialName("paused")
   Paused,

   /** Download was canceled by the user. */
   @SerialName("canceled")
   Canceled,

   /** Download completed. */
   @SerialName("completed")
   Completed,
}
